src/components/Timeline/TimelineEventHeader.tsx

```diff
@@ -26,50 +41,76 @@ const TimelineEventHeader = ({
   secondaryTitle,
   hasPlainDate,
   children,
+  dateNode,
+  tooltip,
+  user,
 }: TimelineEventHeaderProps) => {
   const elements = titleElements?.filter(Boolean) ?? [];
+  const userName = user ? getUserName(user, true) : null;
+
+  const userAttribution =
+    user && userName ? (
+      <Text size={3} color="default2" as="span" marginLeft={1}>
+        by{" "}
+        <Link to={staffMemberDetailsUrl(user.id)} className="timeline-user-link">
+          <Text size={3} color="default2" as="span">
+            {userName}
+          </Text>
+        </Link>
+      </Text>
+    ) : null;
 
   return (
-    <Box
-      display="flex"
-      alignItems="center"
-      flexDirection="row"
-      justifyContent="space-between"
-      width="100%"
-    >
-      {title && (
-        <Text size={3} wordBreak="break-all">
-          {title}
-        </Text>
-      )}
-      {elements.length > 0 && (
-        <Box display="flex" alignItems="center" flexDirection="row" flexWrap="wrap">
-          {elements.map(({ text, link }) => {
-            if (link) {
-              return (
-                <Link to={link} key={`timeline-event-${link}`}>
-                  <Text marginRight={0.5} size={3} color="accent1">
+    <Box display="flex" flexDirection="column" width="100%">
+      <Box
+        display="flex"
+        alignItems="center"
+        flexDirection="row"
+        justifyContent="space-between"
+        width="100%"
+      >
+        <Box display="flex" alignItems="center" flexWrap="wrap">
+          {title && (
+            <Text size={3} color="default1" wordBreak="break-all">
+              {title}
+            </Text>
+          )}
+          {elements.length > 0 && (
+            <Box display="flex" alignItems="center" flexDirection="row" flexWrap="wrap">
+              {elements.map(({ text, link }) => {
+                if (link) {
+                  return (
+                    <Link to={link} key={`timeline-event-${link}`}>
+                      <Text marginRight={0.5} size={3} color="default1" textDecoration="underline">
+                        {text}
+                      </Text>
+                    </Link>
+                  );
+                }
+
+                return (
+                  <Text size={3} color="default1" marginRight={0.5} key={`timeline-event-${text}`}>
                     {text}
                   </Text>
-                </Link>
-              );
-            }
-
-            return (
-              <Text size={3} marginRight={0.5} key={`timeline-event-${text}`}>
-                {text}
-              </Text>
-            );
-          })}
+                );
+              })}
```

Comment:
suggestion: This could be separated into separate component to avoid nested if else logic in a `.map`